### ISSUE TITLE
Add an easy way to use packages inside run_shell 

### ISSUE THREAD

**Issue by fionera** [tweag/rules_nixpkgs#615]
**Is your feature request related to a problem? Please describe.**
As a developer I want to invoke external buildsystems/tools that may do execs themselves. For this I would need a proper PATH with all nixpkgs I want configured. 

**Describe the solution you'd like**
Something like run_shell but with another argument where either one label to a nix_shell_env (bad name i know) which contains all packages, or the list of packages are supplied and fitted into the run_shell environment. Bonus points if include paths etc are correct set

**Describe alternatives you've considered**
Hacking into nix-shell implementation to get a shell file I can use for entering. Doing it manually by including all packages manually, try rules_sh (sadly won't work in my case because of bzlmod dependency trees).

**Additional context**
Basically I want to use rules_nixpkgs as bazel sandbox provider

**Comment by aherrmann** [comment]
Link to corresponding [Bazel Community Slack thread](https://bazelbuild.slack.com/archives/CFB3AE72P/p1732725248232989).
